avocado/utils/pci.py

```
@@ -376,6 +377,87 @@ def get_vendor_id(full_pci_address):
+        raise ValueError(f"Not able to attach {driver} to {device}. Reason: {e}") from e
+
+
+def check_msix_capability(device):
+    """
+    Check whether the PCI device supports Extended Message Signaled Interrupts
+    and if it is currently enabled.
+
+    :param device: full pci address including domain (0000:03:00.0)
+    :return: True if supported, False otherwise
+    """
+    out = process.run(f"lspci -vvs {device}", ignore_status=True, shell=True, sudo=True)
```

Ruff (0.12.2)

Function call with `shell=True` parameter identified, security issue

(S604)